examples/hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/DeviceSimulatorUpdater.java

```diff
@@ -206,12 +205,12 @@ private static boolean isErrorResponse(final UpdateStatus status) {
         private static void handleArtifacts(final String targetToken, final List<UpdateStatus> status,
                 final Artifact artifact) {
 
-            if (artifact.getUrls().containsKey(UrlProtocol.HTTPS)) {
-                status.add(downloadUrl(artifact.getUrls().get(UrlProtocol.HTTPS), targetToken,
-                        artifact.getHashes().getSha1(), artifact.getSize()));
-            } else if (artifact.getUrls().containsKey(UrlProtocol.HTTP)) {
-                status.add(downloadUrl(artifact.getUrls().get(UrlProtocol.HTTP), targetToken,
-                        artifact.getHashes().getSha1(), artifact.getSize()));
+            if (artifact.getUrls().containsKey("https")) {
```

Comment:
`if-block` and `else-if-block` is doing almost the same. Extract it to private method.